### PR TITLE
Keep newlines in the text formatter

### PR DESCRIPTION
**Pull request by hferreiro**
Replace whitespace manually and keep the text newlines. Also, keep the
stories ordering.

### CODE REVIEW

**Comment by guijemont** [comment]
This is a change of behavior that breaks my personal workflow (and maybe others). I suggest adding a command-line option for this (or at least providing a command-line option to be able to have the old behavior).

**Comment by hferreiro** [comment]
Can you share your workflow?

**Comment by guijemont** [comment]
Ok, after checking, it doesn't breaks it, but it looks less good to me. I run with -f markdown and copy-paste the result into our weekly updates tool, as well as save the result locally. I like the format of having "one line per day", but with your patch, I get linebreaks within day entries/items, which is inconsistent with my archive, though the new linebreaks are ignored by the markdown parser, so it's not as bad as I initially thought.

My point remains: I think this is a matter of preference, and therefore should be an option, otherwise I would have to do manual edits to the report every week to make it look the way I like.

**Comment by hferreiro** [comment]
Well, I think that this was clearly a bug in that newlines entered by the user were omitted. But, if keeping the old behaviour is more important, I guess it could be made so that the user can write 2 newlines to enter a real newline. I guess it's up to Martin :-)

**Comment by guijemont** [comment]
IMHO, having the old behavior under a `--ignore-newlines` option is fine, and probably preferable to having to write double newlines, which feels awkward to me. Also, if you want a newline to appear in the markdown result, that means you would have to write 4 newlines in phpreport (since markdown ignores simple newlines)

File: phpreport-report [hunk 1]
```diff
@@ -330,11 +331,20 @@ class DetailedReport(Report):
 
         all_stories = [get_story(task) + task.text for task in tasks_for_day]
 
-        # Many times people add a lot of duplicate descriptions. Just output one of each.
-        all_stories = set(all_stories)
+        # Many times people add a lot of duplicate descriptions. Just output
+        # one of each, keeping the stories ordering
```
Comment (mrobinson) [inline phpreport-report]: Nit: "keeping the stories ordering" should probably read "keeping the stories in the same order."

Comment (mrobinson) [reply phpreport-report]: BTW, this seems like a separate change. Do you mind opening up a PR just for this? I'd like to commit it apart from preserving the newlines.

File: phpreport-report [hunk 2]
```diff
@@ -385,11 +395,14 @@ class TextFormatter(object):
         indent = (first_column_size + 2) * ' '  # Enough to account for the day name offset.
         width = 80 - len(indent)
         for content in contents:
-            second_column = textwrap.fill(content[1],
-                                          break_long_words=False,  # Don't break URLs.
-                                          width=width,
-                                          initial_indent=indent,
-                                          subsequent_indent=indent).strip()
+            second_column = os.linesep.join(
+                [textwrap.fill(line,
```
Comment (mrobinson) [inline phpreport-report]: I'm not sure I understand what is going on here. Are you wrapping each individual line? Is each line really a paragraph?

**Review by mrobinson — COMMENTED** [review]
I think I'm having a bit of trouble understanding the use case for this. Maybe you could give a quick screenshot or text output of the difference between the old and the new approach.

**Comment by mrobinson** [comment]
@hferreiro  So I've been thinking about this and I think that since everyone wants a slightly different output, the best approach here would be for PHPReport to use a template language to produce the reports. That way people could have their own custom templates or simply tweak the existing ones. Would that work for you?

**Pull request closed without merging**